devito/grid.py

```diff
@@ -107,11 +116,51 @@ def __repr__(self):
             self.extent, self.shape, self.dimensions
         )
 
+    @property
+    def extent(self):
+        """Physical extent of the domain in m."""
+        return self._extent
+
+    @property
+    def dtype(self):
+        """Data type inherited by all :class:`Function`s defined on this Grid."""
+        return self._dtype
+
+    @property
+    def origin(self):
+        """Physical coordinates of the domain origin."""
+        return self._origin
+
+    @property
+    def dimensions(self):
+        """Spatial dimensions of the computational domain."""
+        return self._dimensions
+
     @property
     def dim(self):
         """Problem dimension, or number of spatial dimensions."""
         return len(self.shape)
 
+    @property
+    def time_dim(self):
+        """Time dimension associated with this Grid."""
+        return self._time_dim
+
+    @property
+    def stepping_dim(self):
+        """Stepping dimension associated with this Grid."""
+        return self._stepping_dim
+
+    @property
+    def subdomains(self):
+        """The :class:`SubDomain`s defined in this Grid."""
+        return {i.name: i for i in self._subdomains}
```

Comment:
Why `i.name` instead of `i`

Reply:
because I need a dictionary (not a list) from strings to the corresponding objects so that I can easily index into it, even in user code (eg, I'm using it in examples/model.py to retrieve a particular subdomain)

Reply:
Ok makes sense